src/common/kafka/serializer/Frame1DHistogramSerializer.h

```diff
@@ -0,0 +1,178 @@
+// Copyright (C) 2024 European Spallation Source, ERIC. see LICENSE file
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief Implementation of Frame1DHistogramSerializer
+///
+/// See https://github.com/ess-dmsc/streaming-data-types
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "AbstractSerializer.h"
+#include "BinAggregation.h"
+#include "FlatbufferTypes.h"
+#include "common/kafka/Producer.h"
+#include "common/memory/span.hpp"
+
+namespace serializer {
+
+using namespace da00_faltbuffers;
+
+// type trait for type supported as R parameter in the template
+template <class> struct data_type_trait {
+  static constexpr da00_dtype type = da00_dtype::none;
+};
+template <> struct data_type_trait<int8_t> {
+  static constexpr da00_dtype type = da00_dtype::int8;
+};
+template <> struct data_type_trait<int16_t> {
+  static constexpr da00_dtype type = da00_dtype::int16;
+};
+template <> struct data_type_trait<int32_t> {
+  static constexpr da00_dtype type = da00_dtype::int32;
+};
+template <> struct data_type_trait<int64_t> {
+  static constexpr da00_dtype type = da00_dtype::int64;
+};
+template <> struct data_type_trait<uint8_t> {
+  static constexpr da00_dtype type = da00_dtype::uint8;
+};
+template <> struct data_type_trait<uint16_t> {
+  static constexpr da00_dtype type = da00_dtype::uint16;
+};
+template <> struct data_type_trait<uint32_t> {
+  static constexpr da00_dtype type = da00_dtype::uint32;
+};
+template <> struct data_type_trait<uint64_t> {
+  static constexpr da00_dtype type = da00_dtype::uint64;
+};
+template <> struct data_type_trait<float> {
+  static constexpr da00_dtype type = da00_dtype::float32;
+};
+template <> struct data_type_trait<double> {
+  static constexpr da00_dtype type = da00_dtype::float64;
+};
+
+/// \brief A class to handle sending 1-D data collected over a frame to Kafka
+/// using da00 flatbuffers schema
+template <class T, class R = T>
```

Comment:
General note: This file deviates from the ECDC coding standard regarding variable naming.
The current code standard requires variable names to be CamelCase. So 'topic', 'minValue' and '_timeUnit' are 
in violation.